### PR TITLE
Support loading kubeconfig from the 'KUBECONFIG' env

### LINKED ISSUE #3945: Why not support set "--kubeconfig"？ It is not convenience to debug with bin file
https://github.com/pingcap/tidb-operator/blob/5f1c1350fd24e48aeac978b32e2c307647b4f9c9/cmd/controller-manager/main.go#L93

### PR DESCRIPTION
**Pull request by zhucan**
Signed-off-by: zhucan <zhucan.k8s@gmail.com>

### What problem does this PR solve?
In order to debug convenient
Closes #3945 

### What is changed and how does it work?
Add 'KUBE_CONFIG' to env, only set KUBE_CONFIG with "/root/.kube/config" that it can works.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by july2993 — APPROVED** [review]

File: cmd/controller-manager/main.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	"reflect"
 	"syscall"
 
+	"k8s.io/client-go/tools/clientcmd"
+
```
Comment (DanielZhangQD) [inline cmd/controller-manager/main.go:27]: Please remove this empty line.

Comment (zhucan) [reply cmd/controller-manager/main.go:27]: external imports and internal imports should have blank. so not only remove the empty line can solve.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: cmd/controller-manager/main.go [hunk 2]
```diff
@@ -90,7 +92,12 @@ func main() {
 		klog.Info("HELM_RELEASE environment variable not set")
 	}
 
-	cfg, err := rest.InClusterConfig()
+	kubconfig := os.Getenv("KUBE_CONFIG")
```
Comment (DanielZhangQD) [inline cmd/controller-manager/main.go:95]: @zhucan Please use the `KUBECONFIG` env as kubectl uses.

Comment (zhucan) [reply cmd/controller-manager/main.go]: done

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**